src/main/java/org/testng/ReporterConfig.java

```diff
@@ -88,19 +88,19 @@ public static ReporterConfig deserialize(String inputString) {
   /**
    * Creates a reporter based on the current configuration
    */
-  public IReporter newReporterInstance() {
+  public IReporter2 newReporterInstance() {
```

Comment:
You can't change a public api without deprecating it in some versions before.

Could you find a way to keep the previous method (deprecate it if you want) and have your new logic at the same time?